Automattic/Automattic-Tracks-Android
# Experimentation: replace FluxC with internal components

### Description 

This PR replaces usage of `ExperimentsStore` from FluxC, with internal implementation of `AssignmentsRepository` that fetches data from remote, saves it to cache and offer it later.

I aimed for a minimal set of changes here, to not shadow that the logic remains unchanged here.

### How to test

Unit tests are passing after the change. But as they're heavily mocked, we can't really be sure that this caught all possible regressions: https://github.com/Automattic/Automattic-Tracks-Android/pull/239#discussion_r1732906646

Still, this is some safety net. To test this manually, please see #241
